api/src/permissions/AccessMap.ts

```diff
@@ -0,0 +1,38 @@
+import { DocType, Uuid, AclPermission } from "../enums";
+
+/**
+ * Access Map used for access calculations
+ */
+export class AccessMap {
+    /**
+     * Get list of effective access (including inherited access) for the passed group IDs per docType and permission
+     * @param types - Document for which effective access should be calculated
+     * @param permission - Permission for which effective access should be calculated
+     */
+    public calculateAccess: (types: DocType[], permission: AclPermission) => Array<Uuid>;
+
+    public Map: Map<Uuid, Map<DocType, Map<AclPermission, boolean>>>;
```

Comment:
Nitpick: Why does `Map` have a capital letter since it's just a normal public variable?

Reply:
Fixed